### PR TITLE
fix: include commonmark in gatsby-plugin-mdx options

### PR DESCRIPTION
**Pull request by davifantasia**
Closes https://github.com/commercetools/commercetools-docs-kit/issues/770

### CODE REVIEW

File: packages/gatsby-theme-docs/gatsby-config.js [hunk 1]
```diff
@@ -124,6 +124,8 @@ module.exports = (themeOptions = {}) => {
             '.mdx',
             // ".md"
           ],
+          // implement commonmark for stricter compatibility, e.g. backslash transformed to newlines
+          commonmark: true,
```
Comment (emmenko) [inline packages/gatsby-theme-docs/gatsby-config.js:128]: I'm surprised that this works to be honest, as this option is not documented.

https://github.com/gatsbyjs/gatsby/blob/c993d52bff1b2255ba8a2c0d04726bfdb71b070a/packages/gatsby-plugin-mdx/gatsby-node.js#L71-L126

Comment (emmenko) [reply packages/gatsby-theme-docs/gatsby-config.js:128]: image

Comment (emmenko) [reply packages/gatsby-theme-docs/gatsby-config.js:128]: I don't think this is the correct approach.

Comment (nkuehn) [reply packages/gatsby-theme-docs/gatsby-config.js:128]: Outch, this is a regression of a regression we already had during migration.

Functionally, the fix here is correct, the behavior is one of the relevant differences between "tolerant" and commonmark MD parsing. 

The Gatsby devs have just recently introduced their option validation and their issue tracker shows quite some people finding options that are implemented but not part of their validation.   Their plugins often feed options through to sub-plugins but do not document all of them again - a practice that is annoying in the image related plugins, too. 

PS: The MDX / commonmark topic sounds eeerily similar to the remark upgrade to a pure commonmark parser that we discussed recently @emmenko

**Review by nkuehn — APPROVED** [review]
Functionally correct. Please file an issue at gatsbyjs that the options validation does not allow setting the option (or something is wrong with their dependency hell between MDX / remark parser versions that are be default commonmark or not)

File: packages/gatsby-theme-docs/gatsby-config.js (hunk 1 above)
Comment (emmenko) [reply packages/gatsby-theme-docs/gatsby-config.js:128]: Yeah, I tried to follow the trail a bit and it seems indeed that the plugin options are passed down to some other libraries, like `remark-parse`.

Alright then, let's keep it like this for now!

**Review by emmenko — APPROVED** [review]
Let's get this in, thanks for looking into it!

I think we should keep an eye on this though.

**Pull request merged**